publish2cloud.py

```diff
@@ -39,25 +39,12 @@
     REMOTE_SETTINGS_RECORD_PATH = ('/buckets/{bucket_name}'
                                    + '/collections/{collection_name}/records')
     REMOTE_SETTINGS_AUTH = ''
-    if rs_auth_method == "token":
-        if os.environ.get('SHAVAR_REMOTE_SETTINGS_TOKEN', None):
-            REMOTE_SETTINGS_AUTH = BearerAuth((
-                CONFIG.get('main', 'remote_settings_token')
-            ))
-    elif rs_auth_method == "userpass":
-        REMOTE_SETTINGS_AUTH = ('', '')
-        auth_config_exists = (
-            os.environ.get('SHAVAR_REMOTE_SETTINGS_USERNAME', None)
-            and os.environ.get('SHAVAR_REMOTE_SETTINGS_PASSWORD', None)
-        )
-        if auth_config_exists:
-            REMOTE_SETTINGS_AUTH = (
-                CONFIG.get('main', 'remote_settings_username'),
-                CONFIG.get('main', 'remote_settings_password')
-            )
-    else:
-        sys.stderr.write(
-            'No Auth method provided in environment variables\n')
+    if os.environ.get('AUTHORIZATION', None):
+        REMOTE_SETTINGS_AUTH = CONFIG.get('main', 'remote_settings_authorization')
+
+    # This line can be removed once we switch to using kinto-http for requests
+    if rs_auth_method == 'token':
+        REMOTE_SETTINGS_AUTH = BearerAuth(REMOTE_SETTINGS_AUTH)
```

Comment:
I'm sorry I think our documentation is confusing.

When the auth method is user pass, it will be passed as `user:pass` and you need to split it yourself into a tuple for the `requests` auth param:

```
elif rs_auth_method == "userpass":
     REMOTE_SETTINGS_AUTH = tuple(REMOTE_SETTINGS_AUTH.split(":"))
```

See https://github.com/firefox-devtools/remote-settings-mdn-browser-compat-data/blob/d03693d05ebb56ff957538bdf6610fd947248d20/update_remote_settings_records.mjs#L56-L61

and https://requests.readthedocs.io/en/latest/user/authentication/

I'll fix our docs 🙏

Reply:
Just to be sure, in the environment variables the userpass AUTHORIZATION would be something like "username:password" and not "Basic username:password" correct?

Reply:
Yes!